document.go

```diff
@@ -394,38 +423,28 @@ func (d *Document) TranslateIndexToPosition(index int) (row int, col int) {
 	return
 }
 
-// TranslateRowColToIndex given a (row, col), return the corresponding index.
+// TranslateRowColToCursor given a (row, col), return the corresponding cursor.
 // (Row and col params are 0-based.)
-func (d *Document) TranslateRowColToIndex(row int, column int) (index int) {
-	defer debug.Un(debug.Trace("TranslateRowColToIndex", row, column))
-	indexes := d.lineStartIndexes()
-	if row < 0 {
-		row = 0
-	} else if row > len(indexes) {
-		row = len(indexes) - 1
-	}
-	index = indexes[row]
-	line := d.Lines()[row]
-
-	// python) result += max(0, min(col, len(line)))
-	if column > 0 || len(line) > 0 {
-		if column > len(line) {
-			index += len(line)
-		} else {
-			index += column
+func (d *Document) TranslateRowColToCursor(row int, column int) (index int) {
```

Comment:
Since we are dealing with a library, we must keep a backward compatibility of the public API. So it would be better to add a new method `TranslateRowColToCursor` rather than renaming an existing one.

Reply:
Fixed.